test/export/test_sparse.py

```diff
@@ -0,0 +1,229 @@
+    @parametrize("itype", ITYPES)
+    @all_sparse_layouts("layout")
+    def test_idnet(self, dtype, itype, layout):
+        if layout is not torch.sparse_coo:
+            self.skipTest("TODO: support non-coo sparsity!")
+
+        net = IdNet()
+        for sparse_input in self.generate_simple_inputs(
+            layout,
+            device="cpu",
+            dtype=dtype,
+            index_dtype=itype,
+        ):
+            # Build the traced graph.
+            prog = torch.export.export(net, (sparse_input,))
+            # Test arg/output.
+            for i, node in enumerate(prog.graph.nodes):
+                meta = node.meta.get("val", None)
+                if i == 0:
+                    self.assertEqualMeta(meta, sparse_input)
+                else:
+                    self.assertEqual(meta, None)
+
+    @unittest.skipIf(
+        sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+"
+    )
+    @parametrize("dtype", DTYPES)
+    @parametrize("itype", ITYPES)
+    @all_sparse_layouts("layout")
+    def test_sumnet(self, dtype, itype, layout):
+        if layout is not torch.sparse_coo:
+            self.skipTest("TODO: support non-coo sparsity!")
+
+        net = SumNet()
+        for sparse_input in self.generate_simple_inputs(
+            layout,
+            device="cpu",
+            dtype=dtype,
+            index_dtype=itype,
+        ):
+            # Build the traced graph.
+            prog = torch.export.export(net, (sparse_input,))
+            # Test arg/sum/output.
+            for i, node in enumerate(prog.graph.nodes):
+                meta = node.meta.get("val", None)
+                if i == 0:
+                    self.assertEqualMeta(meta, sparse_input)
+                elif i == 1:
+                    self.assertIsInstance(meta, FakeTensor)
+                    self.assertEqual(meta.layout, torch.strided)
+                    self.assertEqual(meta.dtype, dtype)
+                else:
+                    self.assertEqual(meta, None)
+
+    @unittest.skipIf(
+        sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+"
+    )
+    @parametrize("dtype", DTYPES)
+    @parametrize("itype", ITYPES)
+    @all_sparse_layouts("layout")
+    def test_eltwisenet(self, dtype, itype, layout):
+        if layout is not torch.sparse_coo:
+            self.skipTest("TODO: support non-coo sparsity!")
+
+        net = EltwiseNet()
+        for sparse_input in self.generate_simple_inputs(
+            layout,
+            device="cpu",
+            dtype=dtype,
+            index_dtype=itype,
+        ):
+            # Build the traced graph.
+            prog = torch.export.export(net, (sparse_input,))
+            # Test arg/neg/abs/mul/relu/output.
+            for i, node in enumerate(prog.graph.nodes):
+                meta = node.meta.get("val", None)
+                if i <= 4:
+                    self.assertEqualMeta(meta, sparse_input)
```

Comment:
Similarly, `meta` properties must match with the result of `net`, not its input:

```suggestion
                    self.assertEqualMeta(meta, net(sparse_input))
```

Although, in this particular case the properties of result and input are the same, using `net(sparse_input)` exposes the intention of the test better.

Moving the evaluation of `net(sparse_input)` out of the loop would be recommended.